### PR TITLE
Added Movement for player and interaction with movable tile (fixed)

### PR DESCRIPTION
**Pull request by jdamascoty**
- Fixed problem with changes in class files
- Fixed branch name

### CODE REVIEW

File: src/main/java/movement/Movement.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	public void getPlayerCoordinates() {
+		xCoordinatePlayer = gameEngine.getPlayerXCoordinate();
+		yCoordinatePlayer = gameEngine.getPlayerYCoordinate();
+
+		pointOfPlayer = new Point(xCoordinatePlayer, yCoordinatePlayer);
+	}
+
+	public void getTileAfterPlayerCoordinates(String direction) {
```
Comment (borisv13) [inline src/main/java/movement/Movement.java]: Using a String input to decide on behavior is an anti-pattern.
Options:
1. Split into 4 different methods
2. Create an enum for the 4 directions

File: src/main/java/engine/GameEngine.java [hunk 2]
```diff
@@ -16,7 +16,9 @@
 	private final Map<Point, TileType> tiles = new HashMap<>();
 	private int levelHorizontalDimension;
 	private int levelVerticalDimension;
+
 	private Point player;
+	private Point movable;
```
Comment (borisv13) [inline src/main/java/engine/GameEngine.java]: The moment I see this, it indicates that in any one level there will only be a single movable object. Is this true?
If not, get this out of here.

File: src/main/java/movement/Movement.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
+package movement;
+
+import java.awt.Point;
+
+import engine.GameEngine;
+import tiles.TileType;
+
+public class Movement {
```
Comment (borisv13) [inline src/main/java/movement/Movement.java]: There is so much duplication in this class. At least 3 sets of 4 methods that look almost identical to each other.
Extract methods - REFACTOR.

File: src/main/java/ui/TilePainter.java [hunk 4]
```diff
@@ -8,26 +9,59 @@
...
+		}
 		graphics.fillRect(x * tileWidth, y * tileHeight, tileWidth, tileHeight);
 	}
 
 	private void handleTile(Graphics graphics, TileType tileType) {
 		graphics.setColor(TileColorMap.get(tileType));
 	}
 
+	private void handleMovableTile(Graphics graphics, int movableID) {
+		if (movableID == 0) {
```
Comment (borisv13) [inline src/main/java/ui/TilePainter.java]: This looks like poor design. Could have been a map? Or use an enum to map ID -> Color?

File: src/test/java/engine/MovableColorComparisonTest.java [hunk 5]
```diff
@@ -78,7 +78,7 @@ public void check_for_movable_id_one_two_three_four_five_in_order() {
 	public void open_win_message() {
 		SystemWrapper systemWrapper = Mockito.mock(SystemWrapper.class);
 		gameEngine.displayWinMessage(systemWrapper);
-		Mockito.verify(systemWrapper).JOptionPanelDisplay("Congratulations!", "You have successfully aligned blocks!");
```
Comment (jdamascoty) [inline src/test/java/engine/MovableColorComparisonTest.java]: Travis build broke when using JOptionPane to display a win message, so we switched to println instead.

**Pull request closed without merging**